cloud/docker/docker.py

```diff
@@ -1110,6 +1127,15 @@ def get_differing_containers(self):
                 self.reload_reasons.append('volumes_from ({0} => {1})'.format(actual_volumes_from, expected_volumes_from))
                 differing.append(container)
 
+            # LOG_DRIVER
+
+            expected_log_driver = set(self.module.params.get('log_driver') or [])
+            actual_log_driver = set(container['HostConfig']['LogConfig']['Type'] or [])
+            if actual_log_driver != expected_log_driver:
```

Comment:
Ok, I've fixed both 1 and 2.  I think I've probably fixed the false positives for things launched with something other than ansible as well by checking the api version.  That also takes care of worrying about whether `['HostConfig']['LogConfig']['Type']` exists.